homeassistant/components/date/__init__.py

```diff
@@ -0,0 +1,132 @@
+"""Component to allow setting date as platforms."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import date, timedelta
+import logging
+from typing import final
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import FORMAT_DATE
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.config_validation import (  # noqa: F401
+    PLATFORM_SCHEMA,
+    PLATFORM_SCHEMA_BASE,
+)
+from homeassistant.helpers.entity import Entity, EntityDescription
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.typing import ConfigType
+
+from .const import ATTR_DATE, ATTR_DAY, ATTR_MONTH, ATTR_YEAR, DOMAIN, SERVICE_SET_VALUE
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+_LOGGER = logging.getLogger(__name__)
+
+__all__ = ["DOMAIN", "DateEntity", "DateEntityDescription"]
+
+
+async def _async_set_value(entity: DateEntity, service_call: ServiceCall) -> None:
+    """Service call wrapper to set a new date."""
+    return await entity.async_set_value(service_call.data[ATTR_DATE])
+
```

Comment:
This is currently a proxy method. If we have nothing to validate here, we should just omit it.

```suggestion
```

Reply:
decided to keep this... If we use the alternative approach, the attribute name gets passed as a keyword argument which makes `date` a keyword argument to the service method. This would require ugly workarounds where we rename the date module to something else to make it work, this simplifies things.